packages/components/src/public/QueryInfo.test.ts

```diff
@@ -496,4 +496,22 @@ describe('QueryInfo', () => {
             expect(cols[3].readOnly).toBe(true);
         });
     });
+
+    describe('getColumnFromName', () => {
+        test('no name', () => {
+            expect(queryInfo.getColumnFromName(undefined)).toBeUndefined();
+            expect(queryInfo.getColumnFromName("")).toBeUndefined();
+            expect(queryInfo.getColumnFromName(null)).toBeUndefined();
+        });
+
+        test('invalid name', () => {
+            expect(queryInfo.getColumnFromName("nonesuch")).toBeUndefined();
+            expect(queryInfo.getColumnFromName("NAME")).toBeUndefined();
+        });
+
+        test('valid name', () => {
+            const col = queryInfo.getColumnFromName("Name");
+            expect(col.name).toBe("Name");
+        })
```

Comment:
this is likely sufficient, but is it worth adding test cases for the special characters in fieldKey/name case? and for the "found more than one column with name" case?

Reply:
I don't think the "found more than one" case can happen in real life so don't know that a test is warranted, and the logic for finding by name doesn't care about the field key, but I can make sure it won't find the column if you supply the field name.